fideslog/sdk/python/event.py

```diff
@@ -123,3 +123,43 @@ def __init__(
 
         except AssertionError as err:
             raise InvalidEventError(str(err)) from None
+
+
+class UserRegistrationEvent:
+    """
+    A discrete event, representing a user action within a fides tool.
+    """
+
+    def __init__(
+        self,
+        email: str,
+        organization: str,
+        registered_at: datetime,
+    ) -> None:
+        """
+        Define a new user registration event to send to the fideslog server.
+
+        :param email: User email
+        :param organization: User organization
+        :param registered_at: The UTC timestamp when the event occurred, in [ISO 8601](https://en.wikipedia.org/wiki/ISO_8601) format. Must include the UTC timezone, and represent a datetime in the past.
+
+        """
+
+        try:
+            assert email is not None and email.find("@") != -1
```

Comment:
It's not enough to just write the assertion - you also need to provide an explanation for the failure, or the resulting `AssertionError` would be obnoxious to debug. There are examples elsewhere in this file.